src/events/ready.mts

```diff
@@ -7,9 +12,36 @@ export default {
     once: true,
     // Define execution function which in this case is just print out bot user tag
     execute(client: Client) {
-        if (client.user === null) {
+        if (!client.user) {
             throw new Error("client.user is null");
         }
         console.log(`Ready! Logged in as ${client.user.tag}`);
+
+        // Schedule weekly audit log saving:
+        // Saturday @ 11:59 PM
+        cron.schedule("59 23 * * 6", async () => {
+            const guild_id = process.env.DISCORD_GUILD_ID;
+
+            // Check if guild_id is undefined
+            if (!guild_id) {
+                console.error("DISCORD_GUILD_ID environment variable is not set.");
+                return;
+            }
```

Comment:
Assume it's set, or else the bot won't even log in.
```suggestion
```

Reply:
Resolved in a separate commit:
Added `!` after both instances of `guild_id`